templates/generic.php

```diff
@@ -312,6 +364,260 @@
+                    </p>
+                        <?php
+                        $impuestos = $doctoRelacionado->searchNode('pago20:ImpuestosDR');
+                        if (null !== $impuestos) : ?>
+                        <p style="margin: 0px 10px 5px 0;">
+                            <strong>Impuestos Docto Relacionado</strong>
+                        </p>
+                            <?php
+                            $traslados = $impuestos->searchNodes('pago20:RetencionesDR', 'pago20:RetencionDR');
+                            $retenciones = $impuestos->searchNodes('pago20:TrasladosDR', 'pago20:TrasladoDR');
+                            ?>
+                        <p>
+                        <table style="width: 94%">
+                            <tr>
+                                <th style="width: 15%">Tipo</th>
+                                <th style="width: 15%">Base</th>
+                                <th style="width: 15%">Impuesto</th>
+                                <th style="width: 15%">Tipo factor</th>
+                                <th style="width: 20%">Tasa o cuota</th>
+                                <th style="width: 20%">Importe</th>
+                            </tr>
+                            <?php foreach ($traslados as $impuesto) : ?>
+                                <tr>
+                                    <td>Traslado</td>
+                                    <td><?=$this->e($impuesto['BaseDR'])?></td>
+                                    <td><?=$catalagos->catImpuesto($impuesto['ImpuestoDR'])?></td>
+                                    <td><?=$this->e($impuesto['TipoFactorDR'])?></td>
+                                    <td><?=$this->e($impuesto['TasaOCuotaDR'])?></td>
+                                    <td>$<?=$this->e($impuesto['ImporteDR'])?></td>
+                                </tr>
+                            <?php endforeach; ?>
+                            <?php foreach ($retenciones as $impuesto) : ?>
+                                <tr>
+                                    <td>Retención</td>
+                                    <td><?=$this->e($impuesto['BaseDR'])?></td>
+                                    <td><?=$catalagos->catImpuesto($impuesto['ImpuestoDR'])?></td>
+                                    <td><?=$this->e($impuesto['TipoFactorDR'])?></td>
+                                    <td><?=$this->e($impuesto['TasaOCuotaDR'])?></td>
+                                    <td>$<?=$this->e($impuesto['ImporteDR'])?></td>
+                                </tr>
+                            <?php endforeach; ?>
+                        </table>
+                        </p>
+                        <?php endif ?>
+                    <br>
+                    <?php endforeach; ?>
+                <?php endif; ?>
+                <?php $impuestos = $pago20->searchNode('pago20:ImpuestosP');
+                if (null !== $impuestos) : ?>
```

Comment:
It would be clearer if we move upwards this section of Impuestos Pago along with the Pago.

Reply:
Fixed at 2022-09-22